prime-router/src/main/kotlin/azure/WorkflowEngine.kt

```diff
@@ -702,21 +730,14 @@ class WorkflowEngine(
          * These are all potentially heavy weight objects that
          * should only be created once.
          */
-        private val metadata = Metadata.getInstance()
+        private val metadata by lazy { Metadata.getInstance() }
```

Comment:
Context: Adding lazy here allows for using an injected metadata and not having a real metadata catalog being initialized if not needed.